### PR TITLE
Fix metadata worflow

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
**_WHAT DOES THIS PR DO?_**
- Adds automatic metadata .env secret injection
- Renames the datawarehouse folders

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
    - [AN-161](https://airqoteam.atlassian.net/browse/AN-161?atlOrigin=eyJpIjoiNjFkNGEzOTQyNGM3NDFjN2I4MDYyMWFmMjdlMTM3MWQiLCJwIjoiaiJ9)

**_HOW DO I TEST OUT THIS PR?_**
 -Static code analysis

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- n/a

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- n/a

**_ARE THERE ANY RELATED PRs?_**
- n/a

### CODE REVIEW

File: .github/workflows/deploy-previews.yml [hunk 1]
```diff
@@ -696,38 +696,38 @@ jobs:
...
-          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/datawarehouse-pr-previews:${{ github.sha }}
-      
+          cd src/meta-data/
+          docker build --target=deployment --tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/meta-data-pr-previews:${{ github.sha }} .
+          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/pr-previews/meta-data-pr-previews:${{ github.sha }}
+
       - name: Deploy to Cloud Run
         run: |-
-          gcloud run deploy ${{ needs.branch-name.outputs.lowercase }}-datawarehouse-preview \
+          gcloud run deploy ${{ needs.branch-name.outputs.lowercase }}-meta-data-preview \
```
Comment (123MwanjeMike) [inline .github/workflows/deploy-previews.yml:705]: [meta-data deploy preview](https://github.com/airqo-platform/AirQo-api/pull/1078/commits/80c705fc123e22cf2f0c88d50da3e15c9f60e8f6)

File: .github/workflows/remove-deploy-previews.yml [hunk 2]
```diff
@@ -263,39 +263,39 @@ jobs:
...
+
       - name: Setup Cloud SDK
-        uses: 'google-github-actions/setup-gcloud@v0'
-        
+        uses: "google-github-actions/setup-gcloud@v0"
+
       - name: Delete PR deploy preview
         run: |-
-          gcloud run services delete ${{ needs.branch-name.outputs.lowercase }}-datawarehouse \
+          gcloud run services delete ${{ needs.branch-name.outputs.lowercase }}-meta-data \
```
Comment (123MwanjeMike) [inline .github/workflows/remove-deploy-previews.yml:295]: [remove meta-data PR previews](https://github.com/airqo-platform/AirQo-api/pull/1078/commits/96487d8d7e67423a698ff204aa912ce93be363d2)

**Comment by 123MwanjeMike** [comment]
Thanks @123MwanjeMike , this is beautiful!
I will let @NoahNsimbe dig into the specific quality checks later.

As of now, I just have a few comments:

    PR has Git conflicts that need to be resolved.
    Also, I notice the deletion of a directory, what does that mean to any existing proocesses that utilise that directory/ microservice? Have we done some thorough housekeeping ahead of this deletion?

Thanks @Baalmart 
- I have resolved the conflicts
- As for the directory of concern, it was not deleted but rather renamed to [(to-be-deleted)extract-external-data](https://github.com/airqo-platform/AirQo-api/tree/88c399dfcba064adfedb15f41e191355353f98d9/src/data-mgt/python/(to-be-deleted)extract-external-data). All the required migrations were done and its more of an archieve for now. It is for the (old)datawarehouse API that has been replaced by the (new and improved)metadata API. cc @NoahNsimbe

**Review by Baalmart — APPROVED** [review]
LGTM, I will let @NoahNsimbe dig into the nitty gritty details to ensure PR is clean.

**Review by NoahNsimbe — APPROVED** [review]
Thanks @123MwanjeMike ....LGTM....
Please make sure the config maps are up to date.

**Comment by 123MwanjeMike** [comment]
> Thanks @123MwanjeMike ....LGTM....
> Please make sure the config maps are up to date.

Thanks @NoahNsimbe 
I had these updated via secret manager

**Pull request merged**